adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/OAuth2/TokenResponse.cs

```diff
@@ -140,7 +140,9 @@ internal static TokenResponse CreateFromBrokerResponse(IDictionary<string, strin
                 CorrelationId = responseDictionary["correlation_id"],
                 Resource = responseDictionary["resource"],
                 ExpiresOn = long.Parse(responseDictionary["expires_on"].Split('.')[0], CultureInfo.CurrentCulture),
-                ClientInfo = responseDictionary["client_info"]
+                ClientInfo = responseDictionary.ContainsKey("client_info")
+                ? Authenticator.EnsureUrlEndsWithForwardSlash(EncodingHelper.UrlDecode(responseDictionary["client_info"]))
```

Comment:
>Authenticator.EnsureUrlEndsWithForwardSlash(EncodingHelper.UrlDecode [](start = 18, length = 68)

I'm wondering how we are finding this to be needed now. was this fixed in ADAL v3?

Reply:
It was added as part of the cache changes, but broker does not return client_info, so we need to add a check here for that. The next broker release will contain client_info.

Reply:
I was more thinking about the actual change to now url ends with forward slash change. what is the reason for that?

---
In reply to: [197549673](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1101#discussion_r197549673) [](ancestors = 197549673)

Reply:
.. should the same change be done in current ADAL?

---
In reply to: [197550823](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1101#discussion_r197550823) [](ancestors = 197550823,197549673)

Reply:
Will fix this...just need to do a null check. thanks.

Reply:
no because client_info was added was for cache unification, and current broker does not return client_info anyway. Once broker does return client_info we will be on next version of adal anyway.

---
In reply to: [197551638](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1101#discussion_r197551638) [](ancestors = 197551638,197550823,197549673)